packages/dd-trace/src/appsec/remote_config/index.js

```diff
@@ -0,0 +1,259 @@
+'use strict'
+
+const uuid = require('crypto-randomuuid')
+const { EventEmitter } = require('events')
+const Scheduler = require('./scheduler')
+const tracerVersion = require('../../../../../package.json').version
+const request = require('../../exporters/common/request')
+const log = require('../../log')
+
+const clientId = uuid()
+
+const POLL_INTERVAL = 5e3
+
+// There MUST NOT exist separate instances of RC clients in a tracer making separate ClientGetConfigsRequest
+// with their own separated Client.ClientState.
+class RemoteConfigManager extends EventEmitter {
+  constructor (config) {
+    super()
+
+    this.scheduler = new Scheduler((cb) => this.poll(cb), POLL_INTERVAL)
+
+    this.requestOptions = {
+      url: config.url,
+      hostname: config.hostname,
+      port: config.port,
+      method: 'POST',
+      path: '/v0.7/config'
+    }
+
+    this.state = {
+      client: {
+        state: { // updated by `parseConfig()`
+          root_version: 1,
+          targets_version: 0,
+          config_states: [],
+          has_error: false,
+          error: '',
+          backend_client_state: ''
+        },
+        id: clientId,
+        products: [], // updated by `updateProducts()`
+        is_tracer: true,
+        client_tracer: {
+          runtime_id: config.tags['runtime-id'],
+          language: 'node',
+          tracer_version: tracerVersion,
+          service: config.service,
+          env: config.env,
+          app_version: config.version
+        },
+        capabilities: 'AA==' // updated by `updateCapabilities()`
```

Comment:
Maybe do something like this:

```js
const DEFAULT_CAPABILITY = Buffer.from('whatever', 'hex').toString('base64')

// ...

        capabilities: DEFAULT_CAPABILITY // updated by `updateCapabilities()`
```

Otherwise the `'AA=='` string looks very magical!